iroh-gateway/src/handlers.rs

```diff
@@ -49,6 +54,15 @@ use crate::{
     templates::{icon_class_name, ICONS_STYLESHEET, STYLESHEET},
 };
 
+static SUBDOMAINS_REGEX: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(r"(?P<cid>[A-Za-z\d]+)\.(?P<scheme>ip(?:ns|fs))\.(?P<hostname>.*)").unwrap()
```

Comment:
Is this a custom or based on somewhere?

Reply:
in either case, should have definitely some tests just making sure this captures the right things on different inputs

Reply:
Added several tests